applications/sdp/mspi/src/hrt/hrt.c

```diff
@@ -189,3 +201,120 @@ void hrt_write(hrt_xfer_t *hrt_xfer_params)
 		}
 	}
 }
+
+static void hrt_tx_rx(volatile hrt_xfer_data_t *xfer_data, uint8_t frame_width, bool start_counter,
+		      uint16_t cnt0_val, uint16_t cnt1_val)
+{
+	nrf_vpr_csr_vio_shift_ctrl_t shift_ctrl = {
+		.shift_count = BITS_IN_BYTE - 1,
+		.out_mode = NRF_VPR_CSR_VIO_SHIFT_OUTB_TOGGLE,
+		.frame_width = frame_width,
+		.in_mode = NRF_VPR_CSR_VIO_MODE_IN_SHIFT,
+	};
+
+	uint32_t to_send = *((uint32_t *)xfer_data->data);
+
+	if (xfer_data->word_count == 0) {
+		return;
+	}
+
+	nrf_vpr_csr_vio_shift_ctrl_buffered_set(&shift_ctrl);
+
+	for (uint32_t i = 0; i < xfer_data->word_count; i++) {
+		xfer_data->vio_out_set(to_send & MSB_MASK);
+
+		to_send = to_send << BITS_IN_BYTE;
+
+		if ((i == 0) && start_counter) {
+			/* Start both counters */
+			nrf_vpr_csr_vtim_combined_counter_set(
+				(cnt0_val << VPRCSR_NORDIC_CNT_CNT0_Pos) +
+				(cnt1_val << VPRCSR_NORDIC_CNT_CNT1_Pos));
```

Comment:
MDK symbols.

Reply:
I am not sure there is something I can do about it, I need to shift those values. I also need to set them at once, to make sure the timers start at the same time. Should I add some define that will be a "wrapper" for those MDK symbols? Or should I create a ticket for changing the HAL function?

Reply:
Ticket created: https://nordicsemi.atlassian.net/browse/NRFX-7070